backend/search/views.py

```diff
@@ -45,11 +71,23 @@ def search_view(request):
     start_from = data.get('start_from', 0)
     is_analysis = data.get('is_analysis', False)
     variables = data.get('variables', [])
+    behaviour = data.get('behaviour', {})
+
     if is_analysis:
         maximum_results = settings.MAXIMUM_RESULTS_ANALYSIS
     else:
         maximum_results = settings.MAXIMUM_RESULTS
 
+    use_superset = behaviour.get('supersetXpath') is not None
+    # TODO: find a nicer way for running just the superset query
+    use_superset = use_superset and behaviour['supersetXpath'] != xpath
+
+    should_expand_index = behaviour.get('expandIndex', False)
```

Comment:
It seems this is only used if `use_superset` is True, so maybe move inside if block. Also, a comment on what `use_superset` and `should_expand_superset` do would be useful